SU2_CFD/src/numerics/NEMO/convection/ausmplusm.cpp

```diff
@@ -0,0 +1,243 @@
+/*!
+ * \file ausmplusm.cpp
+ * \brief Implementations of the AUSM-family of schemes - AUSM+M.
+ * \author F. Morgado
+ * \version 7.4.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../../include/numerics/NEMO/convection/ausmplusm.hpp"
+
+CUpwAUSMPLUSM_NEMO::CUpwAUSMPLUSM_NEMO(unsigned short val_nDim, unsigned short val_nVar,
+                                           unsigned short val_nPrimVar,
+                                           unsigned short val_nPrimVarGrad,
+                                           CConfig *config): CNEMONumerics (val_nDim, val_nVar, val_nPrimVar, val_nPrimVarGrad,
+                                                          config){
+
+
+  /*--- Allocate data structures ---*/
+  FcL    = new su2double [nVar];
+  FcR    = new su2double [nVar];
+  dmLP   = new su2double [nVar];
+  dmRM   = new su2double [nVar];
+  dpLP   = new su2double [nVar];
+  dpRM   = new su2double [nVar];
+  daL    = new su2double [nVar];
+  daR    = new su2double [nVar];
+  rhos_i = new su2double [nSpecies];
+  rhos_j = new su2double [nSpecies];
+  u_i    = new su2double [nDim];
+  u_j    = new su2double [nDim];
+  Flux   = new su2double[nVar];
+}
+
+CUpwAUSMPLUSM_NEMO::~CUpwAUSMPLUSM_NEMO(void) {
+
+  delete [] FcL;
+  delete [] FcR;
+  delete [] dmLP;
+  delete [] dmRM;
+  delete [] dpLP;
+  delete [] dpRM;
+  delete [] rhos_i;
+  delete [] rhos_j;
+  delete [] u_i;
+  delete [] u_j;
+  delete [] Flux;
+}
+
+CNumerics::ResidualType<> CUpwAUSMPLUSM_NEMO::ComputeResidual(const CConfig *config) {
+
+  unsigned short iDim, iVar, iSpecies;
+  su2double rho_i, rho_j,
+  e_ve_i, e_ve_j, mL, mR, mLP, mRM, mF, pLP, pRM, pF, Phi, sq_veli, sq_velj;
```

Comment:
to echo @pcarruscag s point, let initialize these when used.

Reply:
Done